ewslimeoid.py

```diff
@@ -2761,3 +2778,60 @@ def __init__(
 
 	await resp_cont.post()
 
+async def dress_slimeoid(cmd):
+	user_data = EwUser(member = cmd.message.author)
+	slimeoid = EwSlimeoid(member = cmd.message.author)
+
+	if user_data.life_state == ewcfg.life_state_corpse:
+		response = "Slimeoids don't fuck with ghosts."
+
+	elif slimeoid.life_state == ewcfg.slimeoid_state_none:
+		response = "Wow, great idea! Too bad you don’t even have a slimeoid with which to saturate! You’d think you’d remember really obvious stuff like that, but no. What a dumbass."
+
+	elif slimeoid.life_state == ewcfg.slimeoid_state_forming:
+		response = "Your Slimeoid is not yet ready. Use !spawnslimeoid to complete incubation."
+
+	else:
+		item_search = ewutils.flattenTokenListToString(cmd.tokens[1:])
+		item_sought = ewitem.find_item(item_search = item_search, id_user = cmd.message.author.id, id_server = cmd.message.server.id if cmd.message.server is not None else None)
+
+		cosmetics = ewitem.inventory(
+			id_user = cmd.message.author.id,
+			id_server = cmd.message.server.id,
+			item_type_filter = ewcfg.it_cosmetic
+		)
+
+		# get the cosmetics worn by the slimeoid
+		adorned_cosmetics = []
+		for item in cosmetics:
+			cos = EwItem(id_item = item.get('id_item'))
+			if cos.item_props.get('slimeoid') == 'true':
+				adorned_cosmetics.append(cos)
+
+		if len(adorned_cosmetics) < slimeoid.level:
+			if item_sought != None and item_sought.get('item_type') == ewcfg.it_cosmetic:
+				cosmetic = EwItem(id_item = item_sought.get('id_item'))
+				response = "You "
+
+				# Remove hat
+				if cosmetic.item_props.get('slimeoid') == 'true':
+					response += "take the {} back from {}".format(cosmetic.item_props.get('cosmetic_name'), slimeoid.name)
+					cosmetic.item_props['slimeoid'] = 'false'
+				# Give hat
+				else:
+					# Remove hat from player if adorned
+					if cosmetic.item_props.get('adorned') == 'true':
+						cosmetic.item_props['adorned'] = 'false'
+						response += "take off your {} and give it to {}.".format(cosmetic.item_props.get('cosmetic_name'), slimeoid.name)
```

Comment:
good thinking. it should work the same way when you try to adorn a cosmetic your slimoid is wearing. slimeoid cosmetics should also be dedorned when your slimeoid dies in combat or you dissolve it.

Reply:
done